std/io.zig

```diff
@@ -1276,14 +1281,13 @@ pub fn Deserializer(comptime endian: builtin.Endian, is_packed: bool, comptime E
                         ptr.* = null;
                         return;
                     }
-
+                    
+                    //This should ensure that the optional is set to non-null.
+                    ptr.* = OC(undefined);
                     //The way non-pointer optionals are implemented ensures a pointer to them
```

Comment:
Sorry I missed this before, but this comment is not correct. The layout of non-pointer optionals is not guaranteed. You can get a pointer to the payload in a safe way though:

```zig
ptr.* = OC(undefined); // make it non-null so the following .? is guaranteed safe
const val_ptr = &ptr.?;
```

Reply:
That was derived through experimentation, so my use of the word 'guaranteed' was misleading. Don't know why I didn't think of getting the pointer that way, but it is obviously cleaner.